pkg/relayer/miner/miner.go

```diff
@@ -112,10 +115,10 @@ func (mnr *miner) mapMineRelay(
 	if err != nil {
 		return either.Error[*relayer.MinedRelay](err), false
 	}
-	relayHash := servicetypes.GetHashFromBytes(relayBz)
+	relayHash := protocol.GetHashFromBytes(relayBz)
 
 	// The relay IS NOT volume / reward applicable
-	if uint64(protocol.CountHashDifficultyBits(relayHash)) < mnr.relayDifficultyBits {
+	if bytes.Compare(relayHash[:], mnr.relayDifficultyTargetHash) == 1 {
```

Comment:
Since we have a dynamic per-service difficulty hash, We must fetch difficulty at the beginning of each session.

Reply:
Added a TODO for this in #699